gto-support-okta/src/main/kotlin/org/ccci/gto/android/common/okta/authfoundation/credential/MigrationTokenStorage.kt

```diff
@@ -0,0 +1,46 @@
+package org.ccci.gto.android.common.okta.authfoundation.credential
+
+import com.okta.authfoundation.credential.TokenStorage
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+class MigrationTokenStorage(
+    private val storage: TokenStorage,
+    private val originalStorage: TokenStorage,
+) : TokenStorage, ChangeAwareTokenStorage, TokenStorageObserverRegistry by DefaultTokenStorageObserverRegistry() {
+    private val migrationMutex = Mutex()
+    private var isMigrated = false
+
+    private suspend fun migrateStorage() {
+        if (isMigrated) return
+
+        migrationMutex.withLock {
+            if (isMigrated) return@withLock
+            originalStorage.migrateTo(storage)
```

Comment:
I suppose there is no danger of exceptions here?

Reply:
this would end up throwing the same exceptions any underlying `TokenStorage` would throw when accessing similar methods. Which all of those should be `RuntimeExceptions`

Reply:
there are no migration specific exceptions that get thrown, the logic for the migrateTo method can be found here: https://github.com/CruGlobal/android-gto-support/blob/master/gto-support-okta/src/main/kotlin/org/ccci/gto/android/common/okta/authfoundation/credential/TokenStorage.kt#L5-L9